File: e-commece-multivendor/src/main/java/com/bazar/service/ReviewService.java
package com.bazar.service;

import com.bazar.model.Product;
import com.bazar.model.Review;
import com.bazar.model.User;
import com.bazar.request.CreateReviewRequest;

import java.util.List;

public interface ReviewService {
    Review createReview(CreateReviewRequest req, User user, Product product) throws Exception;

    List<Review> getReviewsByProductId(Long productId);

    Review updateReview(Long reviewId, String reviewText, double reviewRating, List<String> productImages, Long userId)
            throws Exception;

    void deleteReview(Long reviewId, Long userId) throws Exception;

    Review getReviewById(Long reviewId) throws Exception;

    Review getUserReviewForProduct(Long productId, Long userId);
}

File: bazaar-mobile/services/reviewService.js
import api from './api';

export const reviewService = {
    // Get reviews for a product
    getReviews: async (productId) => {
        try {
            const response = await api.get(`/api/products/${productId}/reviews`);
            return response.data;
        } catch (error) {
            throw error;
        }
    },

    // Create a review
    createReview: async (productId, reviewData) => {
        try {
            const response = await api.post(`/api/products/${productId}/reviews`, {
                reviewText: reviewData.reviewText,
                reviewRating: reviewData.reviewRating,
                productImages: reviewData.productImages || [],
            });
            return response.data;
        } catch (error) {
            throw error;
        }
    },

    // Update a review
    updateReview: async (reviewId, reviewData) => {
        try {
            const response = await api.patch(`/api/reviews/${reviewId}`, {
                reviewText: reviewData.reviewText,
                reviewRating: reviewData.reviewRating,
                productImages: reviewData.productImages || [],
            });
            return response.data;
        } catch (error) {
            throw error;
        }
    },

    // Delete a review
    deleteReview: async (reviewId) => {
        try {
            const response = await api.delete(`/api/reviews/${reviewId}`);
            return response.data;
        } catch (error) {
            throw error;
        }
    },
};

export default reviewService;
